lib/bolt/transport/docker.rb

```diff
@@ -16,130 +16,6 @@ def with_connection(target)
         conn.connect
         yield conn
       end
-
-      def upload(target, source, destination, _options = {})
-        with_connection(target) do |conn|
-          conn.with_remote_tmpdir do |dir|
-            basename = File.basename(source)
-            tmpfile = "#{dir}/#{basename}"
-            if File.directory?(source)
-              conn.write_remote_directory(source, tmpfile)
-            else
-              conn.write_remote_file(source, tmpfile)
-            end
-
-            _, stderr, exitcode = conn.execute('mv', tmpfile, destination, {})
-            if exitcode != 0
-              message = "Could not move temporary file '#{tmpfile}' to #{destination}: #{stderr}"
-              raise Bolt::Node::FileError.new(message, 'MV_ERROR')
-            end
-          end
-          Bolt::Result.for_upload(target, source, destination)
-        end
-      end
-
-      def download(target, source, destination, _options = {})
-        with_connection(target) do |conn|
-          download = File.join(destination, Bolt::Util.unix_basename(source))
-          conn.download_remote_content(source, destination)
-          Bolt::Result.for_download(target, source, destination, download)
-        end
-      end
-
-      def run_command(target, command, options = {}, position = [])
-        execute_options = {}
-        execute_options[:tty] = target.options['tty']
-        execute_options[:environment] = options[:env_vars]
-
-        if target.options['shell-command'] && !target.options['shell-command'].empty?
```

Comment:
It looks like this is no longer honored. It seems wrong that it would only have been used for `run_command` and not all `execute` calls though...

Reply:
Good catch! Updated to use this in the `execute` function, so it should be applied to all commands. Also added a test so it'll get caught next time.